core/storage/suggestion/gae_models.py

```diff
@@ -1990,3 +2000,1275 @@ def export_data(
+        language_code: str, contributor_id: str
+    ) -> str:
+        """Constructs a unique ID for a
+        TranslationSubmitterTotalContributionStatsModel instance.
+
+        Args:
+            language_code: str. ISO 639-1 language code.
+            contributor_id: str. User ID.
+
+        Returns:
+            str. An ID of the form:
+
+            [language_code].[contributor_id]
+        """
+        return (
+            '%s.%s' % (language_code, contributor_id)
+        )
+
+    # Here we use MyPy ignore because the signature of this method
+    # doesn't match with BaseModel.get().
+    # https://mypy.readthedocs.io/en/stable/error_code_list.html#check-validity-of-overrides-override
+    @classmethod
+    def get( # type: ignore[override]
+        cls, language_code: str, contributor_id: str
+    ) -> Optional[TranslationSubmitterTotalContributionStatsModel]:
+        """Gets the TranslationSubmitterTotalContributionStatsModel
+        matching the supplied language_code, contributor_id, topic_id.
+
+        Returns:
+            TranslationSubmitterTotalContributionStatsModel|None. The matching
+            TranslationSubmitterTotalContributionStatsModel, or None if no
+            such model instance exists.
+        """
+        entity_id = cls.construct_id(
+            language_code, contributor_id)
+        return cls.get_by_id(entity_id)
+
+    @classmethod
+    def fetch_page(
+        cls,
+        page_size: int,
+        offset: int,
+        sort_by: Optional[str],
+        topic_ids: Optional[List[str]],
+        last_activity: Optional[int],
+        language_code: str
+    ) -> Tuple[Sequence[TranslationSubmitterTotalContributionStatsModel],
+                int,
+                bool]:
+        """Returns the models according to values specified.
+
+        Args:
+            page_size: int. Number of models to fetch.
+            offset: int. Number of results to skip from the beginning of all
+                results matching the query.
+            sort_by: str|None. A string indicating how to sort the result.
+            topic_ids: List[str]|None.
+                List of topics user has contributions.
+            last_activity: int. Number of days within which a user has
+                contributed.
+            language_code: str. Language Code for filter.
+
+        Returns:
+            3-tuple(sorted_results, next_offset, more). where:
+                sorted_results:
+                    list(TranslationSubmitterTotalContributionStatsModel).
+                    The list of models with supplied filters and sort_by.
+                next_offset: int. Number of results to skip in next batch.
+                more: bool. If True, there are (probably) more results after
+                    this batch. If False, there are no further results
+                    after this batch.
+        """
+
+        sort_options_dict = {
+            constants.CD_ADMIN_STATS_SORT_OPTIONS['IncreasingLastActivity']:
```

Comment:
The string value appear to be duplicated, so you should make them constants and put them at the top, e.g. SORT_KEY_INCREASING_LAST_ACTIVITY.

Also, as mentioned in a previous comment, the mapping should be directly from SORT_KEY_INCREASING_LAST_ACTIVITY and you should be passing the sort keys to this function, rather than the human-readable values.

Reply:
Updated. PTAL.